transport-native-epoll/src/main/java/io/netty/channel/epoll/EpollRecvByteAllocatorHandle.java

```diff
@@ -15,26 +15,47 @@
  */
 package io.netty.channel.epoll;
 
+import io.netty.channel.ChannelConfig;
 import io.netty.channel.RecvByteBufAllocator;
 
-abstract class EpollRecvByteAllocatorHandle extends RecvByteBufAllocator.DelegatingHandle {
-    private final boolean isEdgeTriggered;
+class EpollRecvByteAllocatorHandle extends RecvByteBufAllocator.DelegatingHandle {
+    private boolean isEdgeTriggered;
+    private final ChannelConfig config;
     private boolean receivedRdHup;
 
-    public EpollRecvByteAllocatorHandle(RecvByteBufAllocator.Handle handle, boolean isEdgeTriggered) {
+    EpollRecvByteAllocatorHandle(RecvByteBufAllocator.Handle handle, ChannelConfig config) {
         super(handle);
-        this.isEdgeTriggered = isEdgeTriggered;
+        this.config = config;
     }
 
-    public final boolean isEdgeTriggered() {
-        return isEdgeTriggered;
+    final void receivedRdHup() {
+        receivedRdHup = true;
     }
 
-    public final void receivedRdHup() {
-        receivedRdHup = true;
+    boolean maybeMoreDataToRead() {
+        return isEdgeTriggered && lastBytesRead() > 0;
```

Comment:
@trustin @normanmaurer - We may be able to do the same thing for LT mode too, but I figured lets just start with ET where it is necessary and add it as an optimization for LT later if needed.